Use cmpl_only_as_wa_tests list to compile only as w/a

jiejanezhang opened the pull request:
Use cmpl_only_as_wa_tests list to compile only as w/a.

jiejanezhang committed:
Add cmpl_only_as_wa_tests list for compile only

For the current runfail tests, test as compile only.

jiejanezhang committed:
Remove user_defined_rules_2 from cmpl_only_as_wa_tests

user_defined_rules_2 test shall be executed. Remove user_defined_rules_2 from cmpl_only_as_wa_tests.

ShengchenJ commented on features/test_feature.py:
```diff
@@ -221,15 +228,15 @@ def build_test():
     elif re.match('^cufft.*', test_config.current_test) and platform.system() == 'Linux':
         ret = compile_and_link(srcs, cmp_options, objects, link_opts)
-    elif test_config.current_test in cmpl_only_tests:
+    elif test_config.current_test in cmpl_only_tests or test_config.current_test in cmpl_only_as_wa_tests:
```
```suggestion
    elif test_config.current_test in cmpl_only_tests:
```

No need this line.

jiejanezhang replied:
Updated. https://github.com/oneapi-src/SYCLomatic-test/pull/924/commits/ae3d7be4f26bd03bb1f534aeed22597663349753

jiejanezhang committed:
Rename to cmpllink_only_as_wa_tests list.

Rename to cmpllink_only_as_wa_tests list and the tests list can be linked.

jiejanezhang committed:
Add cmpllink_only_as_wa_tests_usmnone for usmnone option

Tests in cmpllink_only_as_wa_tests_usmnone will not be executed under 'option_usmnone_cpu'

jiejanezhang committed:
Consider the case for 'option_usmnone_gpu'

Consider the case for 'option_usmnone_gpu'

jiejanezhang committed:
Simplify the compile only w/a tests handling

Use cmpllink_only_as_wa_tests for all the compile only wa tests under none-cpu options.

jiejanezhang committed:
Do not execute for the tests in cmpllink_only_as_wa_tests

Do not execute for the tests in cmpllink_only_as_wa_tests.
One more correction to the test name in the list.

jiejanezhang committed:
Skip CUDA for get_library_version test

Skip CUDA for get_library_version test.

ShengchenJ reviewed (approved):
LGTM
